cmd/export/main_test.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/exposure-notifications-server/internal/database"
+	"github.com/google/exposure-notifications-server/internal/secrets"
+	"github.com/google/exposure-notifications-server/internal/signing"
+	"github.com/google/exposure-notifications-server/internal/storage"
+)
+
+// Note: this test relies on environment variables and therefore CANNOT be run
+// in parallel.
+func TestSetup(t *testing.T) {
+	t.Run("defaults", func(t *testing.T) {
+		_, databaseConfig := database.NewTestDatabaseWithConfig(t)
+		os.Setenv("DB_NAME", databaseConfig.Name)
+		os.Setenv("DB_USER", databaseConfig.User)
+		os.Setenv("DB_HOST", databaseConfig.Host)
+		os.Setenv("DB_PORT", databaseConfig.Port)
+		os.Setenv("DB_SSLMODE", databaseConfig.SSLMode)
+		os.Setenv("DB_PASSWORD", databaseConfig.Password)
+
+		config, env, closer, err := doSetup()
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer closer()
+
+		// Database
```

Comment:
At some point, I feel like we're testing envconfig here 🤒 , but I'm not sure how else we would have caught the nil pointer bug from #499

Reply:
we could have caught that with unit tests on internal setup

actually that could be sufficient